python_modules/dagster/dagster/_core/definitions/declarative_scheduling/updated_since_cron_condition.py

```diff
@@ -0,0 +1,55 @@
+import datetime
+
+from dagster._utils.schedules import reverse_cron_string_iterator
+
+from .asset_condition import AssetCondition, AssetConditionResult
+from .scheduling_condition_evaluation_context import SchedulingConditionEvaluationContext
+
+
+class UpdatedSinceCronCondition(AssetCondition):
+    cron_schedule: str
+    cron_timezone: str
+
+    @property
+    def condition_description(self) -> str:
+        return f"Updated since latest tick of {self.cron_schedule} ({self.cron_timezone})"
+
+    def _get_previous_cron_tick(
+        self, context: SchedulingConditionEvaluationContext
+    ) -> datetime.datetime:
+        previous_ticks = reverse_cron_string_iterator(
+            end_timestamp=context.effective_dt.timestamp(),
+            cron_string=self.cron_schedule,
+            execution_timezone=self.cron_timezone,
+        )
+        return next(previous_ticks)
+
+    def evaluate(self, context: SchedulingConditionEvaluationContext) -> AssetConditionResult:
+        previous_cron_tick = self._get_previous_cron_tick(context)
+
+        if (
+            # never evaluated
+            context.previous_evaluation is None
+            # partitions def has changed
+            or not context.previous_evaluation.true_subset.is_compatible_with_partitions_def(
```

Comment:
In general I think that anything directly referencing the `AssetSubset` should live on the legacy context.

Reply:
Hm yeah seems reasonable

Reply:
I think it'll be easiest to update this upstack